craft_application/services/provider.py

```diff
@@ -233,3 +237,20 @@ def _capture_logs_from_instance(self, instance: craft_providers.Executor) -> Non
                 emit.debug(
                     f"Could not find log file {source_log_path.as_posix()} in instance."
                 )
+
+    def _setup_instance_bashrc(self, instance: craft_providers.Executor) -> None:
+        """Set up the instance's bashrc to export environment."""
+        bashrc = pkgutil.get_data("craft_application", "misc/instance_bashrc")
+
+        if bashrc is None:
+            emit.debug(
+                "Could not find the bashrc file in the craft-application package"
+            )
+            return
```

Comment:
I am not sure if we should error out or just a warning here

Reply:
I'm not sure either, I'm OK with either.

Reply:
Added the test.